TombEngine/Objects/TR3/Entity/tr3_civvy.cpp

```diff
@@ -79,341 +119,484 @@ namespace TEN::Entities::Creatures::TR3
 		SetAnimation(item, CIVVY_ANIM_IDLE);
 	}
 
+
+	//In the future, if this function works, it could be taken to a more neutral place to be reused by other ally entities.
+	ItemInfo* CivvyFindNearTarget(short itemNumber, std::vector<GAME_OBJECT_ID>& excludedTargetList, int rangeDetection)
+	{
+		auto& item = g_Level.Items[itemNumber];
+		auto& creature = *GetCreatureInfo(&item);
+
+		ItemInfo* result = nullptr;
+
+		auto closetsDistance = FLT_MAX;
+		Vector3 distanceVector;
+		float distanceValue;
+		auto* targetCreature = ActiveCreatures[0];
+		float MaxRange = rangeDetection <= 0 ? FLT_MAX : rangeDetection;
+
+		for (int i = 0; i < ActiveCreatures.size(); i++)
+		{
+			targetCreature = ActiveCreatures[i];
+
+			//Ignore if it's itself, or a non valid Item.
+			if (targetCreature->ItemNumber == NO_ITEM || targetCreature->ItemNumber == itemNumber)
+				continue;
+
+			//Ignore if it's an entity from the Excluded Targets lists.
+			bool forbiddenTarget = false;
+			for (std::vector<GAME_OBJECT_ID>::iterator it = excludedTargetList.begin(); it != excludedTargetList.end(); ++it)
+			{
+				if (g_Level.Items[targetCreature->ItemNumber].ObjectNumber == *it)
+					forbiddenTarget = true;
+			}
+			if (forbiddenTarget)
+				continue;
+
+			//If it's closer than other entity, choose this one.
+			auto& currentItem = g_Level.Items[targetCreature->ItemNumber];
```

Comment:
move it upward at line 144, and check if the target is actually dead with `item.HitPoints <= 0` then use `continue;` if true.

Reply:
you can then replace `g_Level.Items[targetCreature->ItemNumber]` from line 149 to currentItem.ObjectNumber

Reply:
done